examples/step-50/step-50.cc

```diff
@@ -14,906 +14,1449 @@
-          // explained in the @ref mg_paper): We need the remainder of
-          // the operator that we just copied into the
-          // <code>mg_matrices</code> object, namely the part on the
-          // interface between cells at the current level and cells
-          // one level coarser. This matrix exists in two directions:
-          // for interior DoFs (index $i$) of the current level to
-          // those sitting on the interface (index $j$), and the other
-          // way around. Of course, since we have a symmetric
-          // operator, one of these matrices is the transpose of the
-          // other.
-          //
-          // The way we assemble these matrices is as follows: since
-          // the are formed from parts of the local contributions, we
-          // first delete all those parts of the local contributions
-          // that we are not interested in, namely all those elements
-          // of the local matrix for which not $i$ is an interface DoF
-          // and $j$ is not. The result is one of the two matrices
-          // that we are interested in, and we then copy it into the
-          // <code>mg_interface_matrices</code> object. The
-          // <code>boundary_interface_constraints</code> object at the
-          // same time makes sure that we delete contributions from
-          // all degrees of freedom that are not only on the interface
-          // but also on the external boundary of the domain.
-          //
-          // The last part to remember is how to get the other
-          // matrix. Since it is only the transpose, we will later (in
-          // the <code>solve()</code> function) be able to just pass
-          // the transpose matrix where necessary.
+                  (coefficient_value * fe_values.shape_grad(i, q_point) *
+                   fe_values.shape_grad(j, q_point)) *
+                  fe_values.JxW(q_point);
+
+              cell_rhs(i) +=
+                (fe_values.shape_value(i, q_point) * rhs_values[q_point]) *
+                fe_values.JxW(q_point);
+            }
+
+        cell->get_dof_indices(local_dof_indices);
+        constraints.distribute_local_to_global(cell_matrix,
+                                               cell_rhs,
+                                               local_dof_indices,
+                                               system_matrix,
+                                               right_hand_side);
+      }
+
+  system_matrix.compress(VectorOperation::add);
+  right_hand_side.compress(VectorOperation::add);
+}
+
+
+// @sect4{<code>LaplaceProblem::assemble_multigrid()</code>}
+
+// This function assembles and stores the multilevel matrices for the
+// matrix-based GMG method. This function is similar to the one found in
+// step-16, only here it works for distributed meshes. This difference amounts
+// to adding a condition that we only assemble on locally owned cells and a
```

Comment:
I make the mistake often enough to query for `subdomain_id()` rather than `level_subdomain_id()` despite being on the MG levels - do you think it would make this more clear by saying:
```suggestion
// to adding a condition that we only assemble on locally owned level cells and a
```